linuxkit/linuxkit
# Add getty pkg

Signed-off-by: Avi Deitcher <avi@deitcher.net>



**- What I did**

Created a `pkg` called `linuxkit/getty` that replaces the getty functionality from `linuxkit/init`.

**- How I did it**

* Create `pkg/getty/` that processes `/proc/cmdline` options and starts a getty for each console tty.
* respawn logic in the wrapper `rungetty.sh` because getty `exec`s its login (or `/bin/sh`) and expects some wrapper to respawn it when the user exits
* uses `tini -s` is the container root process so that it will reap zombie children
* added `examples/getty.yml`
* `README.md` describing how to use it

The child of `tini` is an `sh` script. It probably could be replaced with a golang binary, but it was easier to get started this way, especially with some of the existing logic taken from `pkg/init/init`.

**Note:** this does *not* remove the logic from `linuxkit/init`; that will be a separate PR to be merged in only after this PR is done.

**- How to verify it**

1. Create an init container that does not have the getty logic in it - sample available on docker hub at `deitch/linuxkit-init:nogetty`
2. Modify `examples/getty.yml` to use that `linuxkit/init` - replace the `init` container line `linuxkit/init:...` with the betty-less init container, e.g. `deitch/linuxkit-init:nogetty`
3. Try each of the 3 function modes and see that it works:
    * no getty service: comment out the entire getty service, see that no login or shell is available
    * insecure: leave the getty service in place, uncomment the env var for `INSECURE`, see that it drops straight to `sh` on console (the old logic)
    * password-secured: leave getty service in place and `INSECURE` commented, see that the console asks for a login, and that `root`/`abcdefgh` (per the example) work to get a shell

Sample getty.yml available at this gist https://gist.github.com/deitch/a24c291321adc6433061618d9fbaf746


**- Description for the changelog**


Create standalone getty container image

**- A picture of a cute animal (not mandatory but encouraged)**

![croc](https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Crocodile_de_Morelet.jpeg/1200px-Crocodile_de_Morelet.jpeg)

OK, it isn't really cute... 

Per discussions with @justincormack and @rneugeba
